06.wc/my_wc.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = ARGV.getopts('l')
+  contents = read_contents
+  display_result(contents, options)
+end
+
+def display_result(contents, options)
+  total_lines = 0
+  total_words = 0
+  total_bytes = 0
+  contents.each do |file_name, content|
+    line_count = content.count
+    total_lines += line_count
+    word_count = content.join.split(/\s+/).count
+    total_words += word_count
+    byte_count = content.join.length
+    total_bytes += byte_count
+    puts adjust_lines_words_bytes(line_count, word_count, byte_count, file_name, options)
+  end
+
+  return if contents.size == 1
+
+  puts adjust_lines_words_bytes(total_lines, total_words, total_bytes, 'total', options)
+end
+
+def adjust_lines_words_bytes(lines, words, bytes, file_or_total, options)
+  result = +''
+  result << format_number(lines)
+  result << "#{format_number(words)}#{format_number(bytes)}" unless options['l']
+  result << " #{file_or_total}"
+
+  # lオプションとwオプションの場合は以下？
+  # result = +''
+  # result << format_number(lines) if options['l']
+  # result << format_number(words) if options['w']
+  # result << "#{format_number(lines)}#{format_number(words)}#{format_number(bytes)}" unless (options['l'] || options['w'])
```

Comment:
```suggestion
  result << format_number(lines) if options['l'] || options.empty?
  result << format_number(words) if options['w'] || options.empty?
  result << format_number(bytes) if options['c'] || options.empty?
```

僕だったらこんな感じに書きますね〜。

説明用変数を導入してもいいかもしれません。

```suggestion
  show_all = options.empty?
  result << format_number(lines) if options['l'] || show_all
  result << format_number(words) if options['w'] || show_all
  result << format_number(bytes) if options['c'] || show_all
```